src/sentry/scripts/similarity/index.lua

```diff
@@ -278,507 +370,334 @@ local function scale_to_total(values)
-                    return table.imap(
-                        results,
-                        function (item)
-                            return {
-                                item[1],
-                                string.format(
-                                    '%f',  -- converting floats to strings avoids truncation
-                                    item[2]
-                                ),
-                            }
-                        end
-                    )
-                end
-            )
-        end
-    ),
-    COMPARE = takes_configuration(
-        function (configuration, arguments)
-            local item_key = arguments[1]
-            local indices = table.slice(arguments, 2)
-
-            local time_series = get_active_indices(
-                configuration.interval,
-                configuration.retention,
-                configuration.timestamp
+        return search(
+            configuration,
+            parameters,
+            limit
+        )
+    end,
+    COMPARE = function (configuration, cursor, arguments)
+        local cursor, limit, item_key, parameters = multiple_argument_parser(
+            argument_parser(validate_integer),
+            argument_parser(validate_value),
+            variadic_argument_parser(
+                object_argument_parser({
+                    {"index", argument_parser(validate_value)},
+                    {"threshold", argument_parser(validate_integer)},
+                })
             )
+        )(cursor, arguments)
 
-            return table.imap(
-                indices,
-                function (index)
-                    local results = fetch_similar(
-                        configuration,
-                        time_series,
-                        index,
-                        fetch_bucket_frequencies(
-                            configuration,
-                            time_series,
-                            index,
-                            item_key
-                        )
-                    )
-
-                    -- Sort the results in descending order (most similar first.)
-                    table.sort(
-                        results,
-                        function (left, right)
-                            return left[2] > right[2]
-                        end
-                    )
-
-                    return table.imap(
-                        results,
-                        function (item)
-                            return {
-                                item[1],
-                                string.format(
-                                    '%f',  -- converting floats to strings avoids truncation
-                                    item[2]
-                                ),
-                            }
-                        end
-                    )
-                end
+        for i, parameter in ipairs(parameters) do
+            parameter.frequencies = get_frequencies(
```

Comment:
This feels a little klugdy

Reply:
ad6344558985d62d0f8219e27483f8e3ab8439b7 - I still don't love this but I feel as though it's an improvement